packages/blocks/src/api/registration.js

```diff
@@ -137,8 +138,7 @@ export function registerBlockType( name, settings ) {
 		return;
 	}
 
-	settings.icon = normalizeIconObject( settings.icon );
```

Comment:
I still think the normalization should happen as soon as possible (registration).

If I take a look at the `normalizeIconObject` function, it only depends on `@wordpress/element` which seems fine for me to be a dependency of `@wordpress/blocks`. Why can't we move it to blocks?

Reply:
Oh I see, it's because it was used in `@wordpress/rich-text` as well? My question is why `registerFormatType` is in RichText? Seems to me that it's better in `blocks` as `rich-text` should be only kept for the richText structure.

For me registering blocks and formats are very related and should be part of the same package.

Reply:
> I still think the normalization should happen as soon as possible (registration).

I don't think that icon's background or shadow color should be a part of the block's registration. It's a very opinionated thing which can differ depending on the UI implementation. This ties this logic with the way `BlockIcon` works. This PR decouples it.

> For me registering blocks and formats are very related and should be part of the same package.

They are very similar but those are 2 completely independent features. You can have a block which doesn't use `RichText` at all or can have multiple `RichText` instances. In my opinion it's only coincidence that one of format options are inline images which is a higher level abstraction which allows enriching UI to put this format also in the inserter.

I also think that we would be able to move out the base functionality of `RichText` to `@wordpress/rich-text` and decouple it from `@wordppress/editor`. In that scenario, you would be able to use it independently of the editor. Having `@wordpress/blocks` in the chain of dependencies wouldn't be expected in that scenario.

Reply:
> They are very similar but those are 2 completely independent features. You can have a block which doesn't use RichText at all or can have multiple RichText instances. 

While I can be on board with this I still think the normalization should happen as soon as possible (the same way we normalize all other block properties at registration).

If we don't want to move `rich-text`  registration to `blocks` we should extract the icon normalization to a package reused in both and it should fix it.

Reply:
Or maybe just duplicate the function as normalizing block icons can have a different meaning than normalizing format icons.